File: Exemplos/App_IntentDigitalHUB_XamarinForms/Elgin_IntentDigitalHUB_XamarinForms-font/E1_Digital_Hub_Xamarin_Forms_M8/IntentServices/Bridge/IniciaCancelamentoVenda.cs
namespace Xamarin_Forms_Intent_Digital_Hub.IntentServices.Bridge
{
    class IniciaCancelamentoVenda : BridgeCommand
    {
        readonly int idTransacao;
        readonly string pdv;
        readonly string valorTotal;
        readonly string dataHora;
        readonly string nsu;

        public IniciaCancelamentoVenda(int idTransacao, string pdv, string valorTotal, string dataHora, string nsu) : base("IniciaCancelamentoVenda")
        {
            this.idTransacao = idTransacao;
            this.pdv = pdv;
            this.valorTotal = valorTotal;
            this.dataHora = dataHora;
            this.nsu = nsu;
        }

        protected override string FunctionParameters()
        {
            return "\"idTransacao\"" + ":" + idTransacao + "," +
                    "\"pdv\"" + ":" + "\"" + pdv + "\"" + "," +
                    "\"valorTotal\"" + ":" + "\"" + valorTotal + "\"" + "," +
                    "\"dataHora\"" + ":" + "\"" + dataHora + "\"" + "," +
                    "\"nsu\"" + ":" + "\"" + nsu + "\"";
        }
    }
}

File: Exemplos/App_IntentDigitalHUB_Ionic/Elgin_IntentDigitalHUB_Ionic-font/src/intentDigitalHubService/BRIDGE/Commands/IniciaCancelamentoVenda.ts
import { BridgeCommand } from "../BridgeCommand";

export class IniciaCancelamentoVenda extends BridgeCommand {
  readonly idTransacao: number;
  readonly pdv: string;
  readonly valorTotal: string;
  readonly dataHora: string;
  readonly nsu: string;

  constructor(
    idTransacao: number,
    pdv: string,
    valorTotal: string,
    dataHora: string,
    nsu: string
  ) {
    super("IniciaCancelamentoVenda");
    this.idTransacao = idTransacao;
    this.pdv = pdv;
    this.valorTotal = valorTotal;
    this.dataHora = dataHora;
    this.nsu = nsu;
  }

  functionParameters(): object {
    return {
      idTransacao: this.idTransacao,
      pdv: this.pdv,
      valorTotal: this.valorTotal,
      dataHora: this.dataHora,
      nsu: this.nsu,
    };
  }
}
